apps/prairielearn/src/lib/user.ts

```diff
@@ -0,0 +1,23 @@
+import { HttpStatusError } from '@prairielearn/error';
+
+/**
+ * Parses a string of UIDs separated by commas, whitespace, line breaks, or semicolons
+ * into an array of unique UIDs.
+ *
+ * @param uidsString The string of UIDs to parse.
+ * @param limit The maximum number of UIDs to accept without throwing an error.
+ */
+export function parseUidsString(uidsString: string, limit: number): string[] {
+  const uids = new Set(
+    uidsString
+      .split(/[\s,;]+/)
+      .map((uid) => uid.trim())
+      .filter((uid) => uid),
+  );
+
+  if (limit != null && uids.size > limit) {
+    throw new HttpStatusError(400, `Cannot provide more than ${limit} UIDs`);
```

Comment:
This is a limit on a single request, so maybe to clarify (in the unlikely case this is triggered):
```suggestion
    throw new HttpStatusError(400, `Cannot provide more than ${limit} UIDs at a time`);
```

Reply:
Done in 692950e.